src/stan/callbacks/stream_writer.hpp

```diff
@@ -68,7 +68,55 @@ class stream_writer : public writer {
     output_ << comment_prefix_ << message << std::endl;
   }
 
+  /**
+   * Writes multiple rows and columns of values in csv format.
+   *
+   * Note: the precision of the output is determined by the settings
+   *  of the stream on construction.
+   *
+   * @param[in] values A matrix of values. The input is expected to have
+   * parameters in the rows and samples in the columns. The matrix is then
+   * transposed for the output.
+   */
+  void operator()(const Eigen::Matrix<double, -1, -1>& values) {
+    output_ << values.transpose().format(CommaInitFmt);
+  }
+  /**
+   * Write a row of values in csv format.
+   *
+   * Note: the precision of the output is determined by the settings
+   *  of the stream on construction.
+   *
+   * @param[in] values A column vector of values.
+   */
+  virtual void operator()(const Eigen::Matrix<double, -1, 1>& values) {
```

Comment:
```suggestion
  void operator()(const Eigen::Matrix<double, -1, 1>& values) {
```

(I think? at least these three overloads should be consistent -- currently one doesn't have this and two do)